core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -222,22 +211,22 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
      * @notice After a passed withdrawal request (i.e., by a call to `requestWithdrawal` and waiting
      * `withdrawalLiveness`), withdraws `positionData.withdrawalRequestAmount` of collateral currency.
      */
-    // TODO: is onlyCollateralizedPosition(msg.sender) correct here? if a position withdraws all their collateral will this still work?
-    // TODO: this currently does not decrement the sponsors oustanding withdrawalRequestAmount. should it?
     // TODO: Decide whether to fold this functionality into withdraw() method above.
-    function withdrawPassedRequest() external onlyPreExpiration() onlyCollateralizedPosition(msg.sender) fees() {
+    function withdrawPassedRequest() external onlyPreExpiration() fees() {
         PositionData storage positionData = _getPositionData(msg.sender);
         require(positionData.requestPassTimestamp < getCurrentTime());
 
         _removeCollateral(positionData.rawCollateral, positionData.withdrawalRequestAmount);
         _removeCollateral(rawTotalPositionCollateral, positionData.withdrawalRequestAmount);
 
-        positionData.requestPassTimestamp = 0;
-
         // Transfer approved withdrawal amount from the contract to the caller.
         collateralCurrency.safeTransfer(msg.sender, positionData.withdrawalRequestAmount.rawValue);
 
         emit RequestWithdrawalExecuted(msg.sender, positionData.withdrawalRequestAmount.rawValue);
+
+        // Reset withdrawal request
+        positionData.withdrawalRequestAmount = FixedPoint.fromUnscaledUint(0);
+        positionData.requestPassTimestamp = 0;
```

Comment:
This is a good catch. I actually noticed this problem when working through the truffle migration process with Regina.